pkg/admission/server.go

```diff
@@ -172,6 +182,36 @@ func handleValidation(request admission.AdmissionRequest) (*admission.AdmissionR
 		} else {
 			ok = true
 		}
+	case v1a1gatewayGVR:
+		var gateway v1alpha1.Gateway
+		deserializer := codecs.UniversalDeserializer()
+		_, _, err := deserializer.Decode(request.Object.Raw, nil, &gateway)
+		if err != nil {
+			return nil, err
+		}
+
+		fieldErr := v1a1Validation.ValidateGateway(&gateway)
+		if fieldErr != nil {
+			message = fmt.Sprintf("%s", fieldErr.ToAggregate())
+			ok = false
+		} else {
+			ok = true
+		}
+	case v1a2gatewayGVR:
+		var gateway v1alpha2.Gateway
+		deserializer := codecs.UniversalDeserializer()
+		_, _, err := deserializer.Decode(request.Object.Raw, nil, &gateway)
+		if err != nil {
+			return nil, err
+		}
+
+		fieldErr := v1a2Validation.ValidateGateway(&gateway)
+		if fieldErr != nil {
+			message = fmt.Sprintf("%s", fieldErr.ToAggregate())
+			ok = false
+		} else {
+			ok = true
+		}
```

Comment:
Please add `gateways` to `resource` in `deploy/admission_webhook.yaml`. Else, this validation will never be invoked.

Reply:
👍